auto_submit/lib/service/validation_service.dart

```diff
@@ -477,36 +481,25 @@ class ProcessMergeResult {
 typedef RetryHandler = Function();
 
 /// Internal wrapper for the logic of merging a pull request into github.
-Future<graphql.QueryResult> _processMergeInternal({
-  required graphql.GraphQLClient client,
-  required Config config,
-  required QueryResult queryResult,
-  required github.PullRequest messagePullRequest,
+Future<github.PullRequestMerge> _processMergeInternal(
+  GithubService githubService,
+  github.RepositorySlug slug,
+  int number,
+  github.MergeMethod mergeMethod, {
+  String? commitMessage,
+  String? requestSha,
 }) async {
-  final String id = queryResult.repository!.pullRequest!.id!;
-
-  final PullRequest pullRequest = queryResult.repository!.pullRequest!;
-  final Commit commit = pullRequest.commits!.nodes!.single.commit!;
-  final String? sha = commit.oid;
-  final int number = messagePullRequest.number!;
-
-  final graphql.QueryResult result = await client.mutate(
-    graphql.MutationOptions(
-      document: mergePullRequestMutation,
-      variables: <String, dynamic>{
-        'id': id,
-        'oid': sha,
-        'title': '${queryResult.repository!.pullRequest!.title} (#$number)',
-      },
-    ),
+  github.PullRequestMerge pullRequestMerge = await githubService.mergePullRequest(
+    slug,
+    number,
+    commitMessage: commitMessage,
+    mergeMethod: mergeMethod,
+    requestSha: requestSha,
   );
 
-  // We have to make this check because mutate does not explicitely throw an
-  // exception, rather it wraps any exceptions encountered.
-  if (result.hasException) {
-    // This exception will bubble up if retries are exhausted.
-    throw RetryableMergeException(result.exception!.graphqlErrors.first.message, result.exception!.graphqlErrors);
+  if (pullRequestMerge.merged == null || !pullRequestMerge.merged!) {
```

Comment:
```suggestion
  if (pullRequestMerge?.merged != true) {
```

Reply:
I'm not sure I understand. The suggestion should be functionally equivalent to your original way, it's just the null safe way of checking this code now.

Reply:
![image](https://user-images.githubusercontent.com/32242716/199614774-ed6720ca-70df-4e33-b0e8-d5797472e3db.png)
I'm saying according to the intellisense that the object cannot be functionally null there so the '?." operator is unnecessary.

Reply:
Ah, I see, I misread your original code

`pullRequestMerge.merged != true` is what you want

Reply:
No problem.